drivers/include/lm75.h

```diff
@@ -0,0 +1,166 @@
+ *
+ * @param[out] dev        device structure to initialize
+ * @param[in] params      initialization parameters
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on initialization related error
+ */
+int lm75_init(lm75_t *dev, const lm75_params_t *params);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Reads the sensor temperature values from TEMP_REG.
+ *
+ * @param[out] dev                device structure
+ * @param[out] temperature        buffer where temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_temperature(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Sets the values for Overtemperature shutdown(OS) and Hysteresis temperature(HYST).
+ * OS gives the temperature's higher bound and HYST the lower bound
+ *
+ * @param[out] dev          device structure
+ * @param[in] temp_os       desired OS temperature in degrees celcius
+ * @param[in] temp_hyst     desired HYST temperature in degrees celcius
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on temperature setting related error
+ */
+int lm75_set_alarm(lm75_t *dev, int temp_hyst, int temp_os);
+
+/**
+ * @brief Overshutdown temperature value of LM75 sensor
+ *
+ * Reads the sensor OS temperature value from TOS_REG.
+ *
+ * @param[out] dev                device structure
+ * @param[out] temperature        buffer where OS temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_os_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Hysteresis temperature value of LM75 sensor
+ *
+ * Reads the sensor hysteresis temperature value from THYST_REG.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where HYST temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_hyst_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Activate the LM75 sensor shutdown mode
+ *
+ * @param[in] dev      device structure to set into shutdown mode
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on mode switching related error
+ */
+int lm75_shutdown(lm75_t *dev);
+
+/**
+ * @brief Dectivate the LM75 sensor shutdown mode
+ *
+ * @param[in] dev      device structure to wake up from shutdown mode
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on mode switching related error
+ */
+int lm75_wake(lm75_t *dev);
+
+/**
+ * @brief Activates one shot conversion mode
+ *
+ * Wakes from shutdown mode, does a single temperature conversion and goes back into shutdown
+ *
+ * @param[in] dev       device structure
+ *
+ * @return M75_ERROR if unsuccessful
+ * @return M75_SUCCESS if successful
+ */
+int tmp1075_one_shot(lm75_t *dev);
```

Comment:
From the description I would have assumed this returns the temperature as well.
Is there a reason for not using one-shot mode in `lm75_get_temperature()` if no alarm is set?

Reply:
It updates the temperature register of the device, but it doesn't return the actual value. To return it the ´lm75_get_temperature´ must be called. 
I will clarify it better in the description

Reply:
I still don't understand why one-shot mode would not be the default if no alarm is configured.